app/src/components/collectionStructure/collectionStructure.tsx

```diff
@@ -1,258 +1,380 @@
-import { mockCollectionChildrenResponse } from "__tests__/__mocks__/data/mockCollectionStructure";
-import { useScrollIntoViewIfNeeded } from "@/src/hooks/useScrollIntoViewIfNeeded";
+import { SearchManager } from "@/src/utils/searchManager";
 import { headerBreakpoints } from "@/src/utils/breakpoints";
+import AccordionTree from "./accordionTree";
+import { CARDS_PER_PAGE } from "@/src/config/constants";
 
-export interface CollectionChildProps {
+export interface OpenStateItem {
   title: string;
-  itemCount: string;
-  children?: CollectionChildProps[];
-  hasChildren: boolean;
   uuid: string;
-}
-
-interface OpenStateItem {
-  title: string;
   level: number;
+  itemCount: string | number;
   isOpen: boolean;
+  hasSubContainers: boolean;
+  children?: OpenStateItem[];
 }
 
-const ButtonText = ({ title, hasChildren, level }) => {
-  let text = (
-    <Text
-      flex="1"
-      marginBottom="0"
-      whiteSpace="nowrap"
-      overflow="hidden"
-      textOverflow="ellipsis"
-      paddingLeft={hasChildren ? "s" : "28px"}
-      fontWeight="500"
-    >
-      {title}
-    </Text>
-  );
-  let truncationLength = 30 - level * 7;
-  if (title.length > truncationLength) {
-    return (
-      <Tooltip zIndex="1000" content={text}>
-        {text}
-      </Tooltip>
-    );
+interface CollectionStructureProps {
+  uuid: string;
+  updateURL: (queryString: string) => Promise<void>;
+  searchManager: SearchManager;
+}
+
+const fetchChildren = async (uuid: string): Promise<OpenStateItem[]> => {
+  const allChildren: OpenStateItem[] = [];
+  let page = 1;
+  while (true) {
+    const res = await fetch(`/api/collectionchildren/${uuid}?page=${page}`);
+    if (!res.ok) throw new Error("Failed to fetch children");
+    const data = await res.json();
+    const mappedChildren = data.children.map((child) => ({
+      title: child.title,
+      uuid: child.uuid,
+      itemCount: child.itemCount,
+      hasSubContainers: child.hasSubContainers,
+      level: 0,
+      isOpen: false,
+      children: [],
+    }));
+    allChildren.push(...mappedChildren);
+
+    if (data.children.length < CARDS_PER_PAGE) break;
+    page++;
   }
-  return text;
+
+  return allChildren;
 };
 
-const fetchChildren = async (
-  title: string
-): Promise<CollectionChildProps[]> => {
-  return new Promise((resolve) => {
-    setTimeout(() => {
-      // simulating fetching real data but actually just filtering the mock
-      const findChildren = (data: CollectionChildProps[], title: string) => {
-        for (const item of data) {
-          if (item.title === title) {
-            return item.children || [];
-          } else if (item.children) {
-            const result = findChildren(item.children, title);
-            if (result.length > 0) return result;
+const closeAllChildren = (node: OpenStateItem): OpenStateItem => ({
+  ...node,
+  isOpen: false,
+  children: node.children?.map(closeAllChildren) ?? [],
+});
+
+const toggleItem = async (
```

Comment:
We're (DXP/DS) thinking as a general rule "three or more args means it should be an arg object" to remove positional args.

Reply:
This function is doing a lot more than just toggling an item... This is quite complicated and reading through it, even with comments, it's a bit difficult to follow. Sorry, about the lack of better feedback but won't be able to dig into this too deeply. 

Quick suggestions are to rename the function to include what else it does, and remove the inner functions out.